ui/page/dexclient/create_wallet_modal.go

```diff
@@ -103,67 +104,113 @@ func (md *createWalletModal) OnResume() {
 	}
 }
 
-func (md *createWalletModal) Handle() {
-	if md.cancel.Button.Clicked() && !md.isSending {
-		md.Dismiss()
+func (md *createWalletModal) SetRegisterAction(registerAction bool) *createWalletModal {
+	md.isRegisterAction = registerAction
+	return md
+}
+
+func (md *createWalletModal) validateInputs(isRequiredWalletPassword bool) (bool, string, string) {
+	appPass := md.dexClientPassword
+	if appPass == "" {
+		appPass = md.appPassword.Editor.Text()
 	}
 
-	if md.submit.Button.Clicked() {
-		if md.appPassword.Editor.Text() == "" || md.isSending {
-			return
-		}
+	if appPass == "" {
+		md.submitBtn.SetEnabled(false)
+		return false, "", ""
+	}
 
-		md.isSending = true
-		md.modal.SetDisabled(true)
+	wallPassword := md.walletPassword.Editor.Text()
+	if isRequiredWalletPassword && wallPassword == "" {
+		md.submitBtn.SetEnabled(false)
+		return false, "", ""
+	}
 
-		go func() {
-			defer func() {
-				md.isSending = false
-				md.modal.SetDisabled(false)
-			}()
+	md.submitBtn.SetEnabled(true)
+	return true, appPass, wallPassword
+}
 
-			coinID := md.walletInfoWidget.coinID
-			coinName := md.walletInfoWidget.coinName
-			if md.Dexc().HasWallet(int32(coinID)) {
-				md.Toast.NotifyError(fmt.Sprintf("already connected a %s wallet", coinName))
-				return
-			}
+func (md *createWalletModal) Handle() {
+	isRequiredWalletPassword := md.walletInfoWidget.coinID == dcr.BipID
+	canSubmit, appPass, walletPass := md.validateInputs(isRequiredWalletPassword)
 
-			settings := make(map[string]string)
-			var walletType string
-			appPass := []byte(md.appPassword.Editor.Text())
-			walletPass := []byte(md.walletPassword.Editor.Text())
-
-			switch coinID {
-			case dcr.BipID:
-				selectedAccount := md.sourceAccountSelector.SelectedAccount()
-				settings[dcrlibwallet.DexDcrWalletIDConfigKey] = strconv.Itoa(selectedAccount.WalletID)
-				settings["account"] = selectedAccount.Name
-				settings["password"] = md.walletPassword.Editor.Text()
-				walletType = dcrlibwallet.CustomDexDcrWalletType
-			case btc.BipID:
-				walletType = "SPV" // decred.org/dcrdex/client/asset/btc.walletTypeSPV
-				walletPass = nil   // Core doesn't accept wallet passwords for dex-managed spv wallets.
+	if isWalletPasswordSubmit, _ := decredmaterial.HandleEditorEvents(md.walletPassword.Editor); isWalletPasswordSubmit {
+		if md.dexClientPassword != "" && canSubmit {
```

Comment:
Don't need this duplicate check, if `canSubmit` is true, then a dex client password is provided.
```suggestion
		if canSubmit {
```